drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -157,7 +157,7 @@ static void bt_ipm_rx_thread(void)
 
 		hcievt = k_fifo_get(&ipm_rx_events_fifo, K_FOREVER);
 
-		k_sem_take(&ipm_busy, K_NO_WAIT);
+		k_sem_take(&ipm_busy, K_FOREVER);
```

Comment:
It looks like you have duplicate `k_sem_give` in the while loop here. I think maybe the ones in the switch statement can be removed.

Reply:
Thanks, actually it seems that the previous change broke the error management in this loop.
I've pushed a fix